src/components/Articles.jsx

```diff
@@ -9,57 +9,136 @@ export default function Articles() {
-                            break;
-                        case "coding":
-                            photoSrc = codingPhoto;
-                            photoAlt = "messy computer code";
-                            break;
-                        case "football":
-                        // if anything goes awry, the default will render a football photo
-                        default:
-                            photoSrc = footballPhoto;
-                            photoAlt = "football stadium";
-                            break;
-                    }
-                    return (
-                        <li key={article.article_id}>
-                            <Link to={`/articles/${article.article_id}`}>
-                                <img src={photoSrc} alt={photoAlt} />
-                                <br />
-                                <strong>{article.title}</strong> <i>by</i>{" "}
-                                {article.author}
-                                <br />
-                                <span className="Articles--topic">
-                                    #{article.topic}
-                                </span>
-                            </Link>
-                        </li>
-                    );
-                })
-            ) : (
-                <p>Loading articles...</p>
-            )}
-        </ul>
+        <main className="Articles">
+            <select onChange={handleSort} ref={ref}>
+                <option>Sort articles by...</option>
+                <option>Date (ascending)</option>
+                <option>Date (descending)</option>
+                <option>Title (ascending)</option>
+                <option>Title (descending)</option>
+                <option>Author (ascending)</option>
+                <option>Author (descending)</option>
+                <option>Comments (ascending)</option>
+                <option>Comments (descending)</option>
+            </select>
+            <ul>
+                {!isLoading ? (
+                    articles.map((article) => {
+                        let photoSrc, photoAlt;
+                        switch (article.topic) {
+                            case "cooking":
+                                photoSrc = cookingPhoto;
+                                photoAlt = "colourful cookery";
+                                break;
+                            case "coding":
+                                photoSrc = codingPhoto;
+                                photoAlt = "messy computer code";
+                                break;
+                            case "football":
+                            // if anything goes awry, the default will render a football photo
+                            default:
+                                photoSrc = footballPhoto;
+                                photoAlt = "football stadium";
+                                break;
+                        }
```

Comment:
I like the use of the switch statement here although I probably would have a much more neutral default photo, maybe even one that explains there's been an issue with retrieving that image data.

Reply:
I've now replaced it with a placeholder image